delete_me_proxies_example/HumanReadableCleaningSteps.Rmd

```diff
@@ -0,0 +1,118 @@
+---
+output:
+  pdf_document: default
+  html_document: default
+---
+
+# Rudimentary cleaning steps applied to data
+
+<br>
+
+We applied this to all of the data for METALS, HABS, and PFAS that is summarized in Task 1.1. The ancillary data summaries in Task 1.1. did not have this cleaning applied to them. For the purposes of Sam's analysis, I did pass the ancillary data that matched metals data through these cleaning steps.
+
+<br>
+
+1. **Missing values.** If either the `ResultMeasureValue` or `DetectionQuantitationLimitMeasure.MeasureValue` columns were NA, treat the data as missing. Also, if the `ResultDetectionConditionText` column said some variant of "not reported" or the `ResultCommentText` said "not collected"" or "not analyzed"", treat the data as missing. Then, remove any of the missing observations.
+2. **Zero values.** If the data value is zero and there was not enough information to resolve using the `DetectionQuantitationLimitMeasure.MeasureValue` column (meaning that field was also zero or missing), then remove the observation.
+3. **Handling non-detects.** First decide that a value is below or above the detection limit when the `ResultMeasureValue` column is missing but the `DetectionQuantitationLimitMeasure.MeasureValue` is not missing AND the text in `ResultDetectionConditionText` matches "*Non-detect", "Not Detected", or "Present Below Quantification Limit" (meaning value is below the detection limit) OR `ResultDetectionConditionText` matches "Present Above Quantification Limit" (meaning the value is above the detection limit). We added a new column called `resultVal` for the result value to reflect non-detect data. If the observation was determined to be a non-detect, the `resultVal` column will be filled with the value in `DetectionQuantitationLimitMeasure.MeasureValue` but if not, then `resultVal` will be filled with `ResultMeasureValue`. We also add a new column for the units called `resultUnits`. If the observation was determined to be a non-detect, the `resultUnits` column will be filled with the value from `DetectionQuantitationLimitMeasure.MeasureUnitCode` but if not, then `resultUnits` will be filled with the value from `ResultMeasure.MeasureUnitCode`. Finally, we added a column called `remarkCd`. This column was filled with `<` when the observation was determined to be below the detection limit and `>` when the observation was determined to be above the detection limit. If the value was not a non-detect, then the `remarkCd` column was filled with `NA`.
```

Comment:
I like the three munging tasks outlined here - I would consider these three steps here to be an extremely "light touch" on the data and I think we should use the same approach in our example pipeline.

Reply:
I agree, @padilla410 - handling missing values, zero values, and non detects likely represent very common problems for folks trying to use WQP data. If it's helpful, we can also compare specific mungeing steps or look for inspiration within the DRB WQP harmonization [readme](`DRB.WQP.harmonization.code.readme.pdf`) and [running notes](https://code.usgs.gov/wma/iws/wq-data-harmonization/-/blob/main/DP2_Documentation.md) (both of those link to an internal GitLab repo).

Reply:
I think non-detects could get us into some murky waters. There are a lot of ways to handle them and varying levels of sophistication in approaches. It might be save us time to steer clear of this and keep this pipeline scoped to the more straightforward cleaning steps.